src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpointImpl.java

```diff
@@ -0,0 +1,240 @@
+        for (EntityType entityType : entityTypesOfLayer) {
+            JsonObject jsonEntityType = getEntityTypeJson(entityType);
+
+            jsonEntityTypesOfLayer.add(jsonEntityType);
+            getChildJsons(tree.get(entityType.getId()), layerMap, layerId + 1, tree);
+        }
+
+        if (layerMap.containsKey(layerId)) {
+            List<JsonObject> otherTypesOfLayer = layerMap.get(layerId);
+            layerMap.remove(layerId, otherTypesOfLayer);
+            jsonEntityTypesOfLayer.addAll(otherTypesOfLayer);
+            layerMap.put(layerId, jsonEntityTypesOfLayer);
+        } else {
+            layerMap.put(layerId, jsonEntityTypesOfLayer);
+        }
+    }
+
+    /**
+     * This method returns the given entity type as json object.
+     * @param entityType the entity type to be processed
+     * @return json representation of the given entity type
+     */
+    private JsonObject getEntityTypeJson(EntityType entityType) {
+        JsonObject jsonEntityType = new JsonObject();
+        jsonEntityType.addProperty("Id", entityType.getId());
+        jsonEntityType.addProperty("ParentId", entityType.getParentId());
+        jsonEntityType.addProperty("Name", entityType.getName());
+        return jsonEntityType;
+    }
+
+    /**
+     * This method returns the given mappings as json array.
+     * @param mappings all mappings to be processed
+     * @return json representation of the given mappings
+     */
+    private JsonArray getMappingsJson(List<EventEntityMapping> mappings) {
+        JsonArray mappingsJson = new JsonArray();
+
+        for (EventEntityMapping mapping : mappings) {
+            JsonObject jsonMapping = new JsonObject();
+            jsonMapping.addProperty("Id", mapping.getId());
+            jsonMapping.addProperty("EventTypeId", mapping.getEventTypeId());
+            jsonMapping.addProperty("EntityTypeId", mapping.getEntityTypeId());
+            List<MappingCondition> conditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mapping.getId());
+            jsonMapping.add("EventEntityMappingConditions", getMappingConditionsJson(conditions));
+        }
+
+        return mappingsJson;
+    }
+
+    /**
+     * This method returns the given conditions as json array.
+     * @param conditions the conditions to be processed
+     * @return json representation of the given mapping conditions
+     */
+    private JsonArray getMappingConditionsJson(List<MappingCondition> conditions) {
+        JsonArray jsonConditions = new JsonArray();
+
+        for (MappingCondition condition : conditions) {
+            JsonObject jsonCondition = new JsonObject();
+            jsonCondition.addProperty("EventTypeAttributeId", condition.getEventTypeAttributeId());
+            jsonCondition.addProperty("EntityTypeAttributeId", condition.getEntityTypeAttributeId());
+            jsonConditions.add(jsonCondition);
+        }
+        
+        return jsonConditions;
+    }
+
+    /**

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadoc_Style_BP2016)